lib/experimental/fonts-api/fonts-api.php

```diff
@@ -176,13 +176,15 @@ function wp_register_font_provider( $name, $classname ) {
 	 *
 	 * @since X.X.X
 	 *
-	 * @param string|string[]|false $handles Optional. Items to be processed: queue (false),
-	 *                                       single item (string), or multiple items (array of strings).
-	 *                                       Default false.
+	 * @param string|string[]|false $handles            Optional. Items to be processed: queue (false),
+	 *                                                  single item (string), or multiple items (array of strings).
+	 *                                                  Default false.
+	 * @param bool                  $for_iframed_editor Optional. Whether the fonts are for iframed editor.
+	 *                                                  Default false.
 	 * @return array|string[] Array of font handles that have been processed.
 	 *                        An empty array if none were processed.
 	 */
-	function wp_print_fonts( $handles = false ) {
+	function wp_print_fonts( $handles = false, $for_iframed_editor = false ) {
```

Comment:
This function needs to know if the request to print is coming from the iframed editor for decision making about which fonts to print.

What strategies could be used?

- Approach 1: Adding a parameter to the function's signature

Pros:
- straight-forward approach as the invoking code in the script-loader files sets it to `true`.
- avoids the overhead of firing a hook.
- avoids third-party hooking into this part of the code, i.e. a hook is available later in the call stack.
- more performant than the other approaches.

Cons:
- Deviates from `wp_print_styles()` and `wp_print_scripts()`.
- Adds a method parameter which must be maintained for BC once in WP Core.

* Approach 2: Hook into the `'block_editor_settings_all'` to modify the `'__unstableResolvedAssets`' setting.

In WP Core, the call stack is this:

`_wp_get_iframed_editor_assets()`
invokes 
`_wp_get_iframed_editor_assets()`
which will invoke
`wp_print_fonts()`

There is a filter after this code called `'block_editor_settings_all'`.

Gutenberg uses this approach to modify the settings for the upcoming release.

Pros:
* No additional function parameter is needed, meaning no future BC concerns for adding a parameter.

Cons:
* Once in Core, the code would be separate from `_wp_get_iframed_editor_assets()`, which would make maintaining more complex.
* Contributors would need to stay aware of changes in `_wp_get_iframed_editor_assets()` that could require changes within `wp_print_fonts()` of which setting or subsetting to modify. (Conversely if invoking directly within `_wp_get_iframed_editor_assets()` all knowledge is contained in one place for the iframed editor assets.)
* Performance hit: adding and firing a hooked callback has a performance cost.

* Approach 3: Add a hook in `_wp_get_iframed_editor_assets()`.

Pros:
* Wires internal and external iframed editor asset functionality together (vs using a hook later in the call stack).

Cons:
* Adds another hook that must be maintained forever due to BC.
* Performance hit: firing a hook has a performance cost.